pkg/kf/commands/apps/push.go

```diff
@@ -153,36 +168,40 @@ func NewPushCommand(p *config.KfParams, client apps.Client, pusher apps.Pusher,
 				}
 
 				var imageName string
+				if containerImage == "" {
+
+					srcPath := filepath.Join(path, app.Path)
+					switch {
+					case sourceImage != "":
+						imageName = sourceImage
+					default:
+						imageName = apps.JoinRepositoryImage(containerRegistry, apps.SourceImageName(p.Namespace, app.Name))
+
+						if err := b.BuildSrcImage(srcPath, imageName); err != nil {
+							return err
+						}
+					}
 
-				srcPath := filepath.Join(path, app.Path)
-				switch {
-				case sourceImage != "":
-					imageName = sourceImage
-				default:
-					imageName = apps.JoinRepositoryImage(containerRegistry, apps.SourceImageName(p.Namespace, app.Name))
-
-					if err := b.BuildSrcImage(srcPath, imageName); err != nil {
+					// Read environment variables from cli args
```

Comment:
This seems to imply that if you set --docker-image, it won't bring in your environment variables.

Reply:
good catch, should be fixed. Other lags might not work though